kpi/utils/mongo_helper.py

```diff
@@ -61,8 +183,135 @@ def _is_attribute_encoded(cls, key):
         :return: string
         """
         return (
-            key not in cls.KEY_WHITELIST and (
-                key.startswith('JA==') or
+                key not in cls.KEY_WHITELIST and (
+                    key.startswith('JA==') or
                     key.count('Lg==') > 0
-            )
+                )
         )
+
+    @staticmethod
+    def _is_nested_reserved_attribute(key):
+        """
+        Checks if key starts with one of variables values declared in NESTED_MONGO_RESERVED_ATTRIBUTES
+
+        :param key: string
+        :return: boolean
+        """
+        for reserved_attribute in NESTED_MONGO_RESERVED_ATTRIBUTES:
+            if key.startswith(u"{}.".format(reserved_attribute)):
+                return True
+        return False
+
+    @classmethod
+    def get_instances(cls, mongo_userform_id, hide_deleted=True, **kwargs):
+
+        params = cls.validate_params(**kwargs)
+        start = params.get("start")
+        limit = params.get("limit")
+        sort = params.get("sort")
+        fields = params.get("fields")
+        query = params.get("query")
+        instances_ids = params.get("instances_ids")
+
+        # check if query contains and _id and if its a valid ObjectID
+        if "_uuid" in query and ObjectId.is_valid(query.get("_uuid")):
+            query["_uuid"] = ObjectId(query.get("_uuid"))
```

Comment:
I wonder if it would be better to return no results if `is_valid()` fails. Then again, I don't know what `is_valid()` does

Reply:
https://github.com/mongodb/mongo-python-driver/blob/master/bson/objectid.py#L147-L163
`is_valid()` takes a `str` or an `ObjectId` as parameter. 

These lines are only there to validate that `_uuid` is a valid Mongo unique identifier if it's present. 
When users filter the submissions through the API (e.g. `/assets/{uid}/submissions/?query=...`), they don't provided a `_uuid` but need the results to be returned. 

IMHO, we should leave this as is.